### PR TITLE
feat: Deploy Eclipse Che on a virtual Kubernetes cluster 

### PR DESCRIPTION
**Pull request by maheshrajrp**
## What does this pull request change?
Addition of Support for Che with vCluster

## What issues does this pull request fix or reference?
https://github.com/eclipse/che/issues/22845

## Specify the version of the product this pull request applies to
7.82

## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [x] Successfully tested.
- Any page or link rename:
  - [x] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [x] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [x] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

**Comment by maheshrajrp** [comment]
@tolusha @deerskindoll @ibuziuk reopened PR with the latest changes.

**Comment by tolusha** [comment]
@deerskindoll 
Could you confirm that the article has to be split into smaller ones? 
If so, I can help with that.

this is a question for @maheshrajrp.

**Comment by maheshrajrp** [comment]
> @deerskindoll 
> Could you confirm that the article has to be split into smaller ones? 
> If so, I can help with that.
> 
> this is a question for @maheshrajrp.

hi @tolusha 
im not sure if we could split this, the best we could do is move just the keycloak setup. the client id, secret creation are still needed. i fear it might do more harm than it's worth it ?

in any case, do you have thoughts or suggestions @tolusha @deerskindoll ?

**Comment by deerskindoll** [comment]
I'm worried about the readability and clarity of the process. It's really long and involves many components that go beyond the process of installation. I'd prefer to shift keyclok setup to a separate process and add it as a prerequisite to the installation itself. Let me check with another writer for a second opinion.

@maheshrajrp I need you to include the fix for the missing attribute issue, the build and verify PR workflow won't passs otherwise. There's also a potential issue with the usage of `[source,shell]` but I need to look into it more.

**Comment by maheshrajrp** [comment]
> I'm worried about the readability and clarity of the process. It's really long and involves many components that go beyond the process of installation. I'd prefer to shift keyclok setup to a separate process and add it as a prerequisite to the installation itself. Let me check with another writer for a second opinion.
> 
> @maheshrajrp I need you to include the fix for the missing attribute issue, the build and verify PR workflow won't passs otherwise. There's also a potential issue with the usage of `[source,shell]` but I need to look into it more.

@deerskindoll can you give some hints on what you mean by attributes issues or point me to some documentation? it would really help.. first-time contributor here, sorry and thanks !!

**Comment by deerskindoll** [comment]
No worries. I left the suggestion in your original PR. `{domain_name}` and `{keycloak_host}` are causing an asciidoc issue because asciidoc treats them as attributes. 

You need to add a backslash `\` to avoid this error: `\{domain_name}` and {\{keycloak_host}`

**Comment by deerskindoll** [comment]
There are two more issues with the code blocks:

- `[source, shell]` https://redhat-documentation.github.io/supplementary-style-guide/#code-example-syntax-highlighting
- removing the `$` prefix https://redhat-documentation.github.io/supplementary-style-guide/#commands-with-root-privileges

Can you please fix these issues?

**Comment by maheshrajrp** [comment]
@deerskindoll . Thanks for the hints. I can understand. Will fix those issues. Give me a day or two (as right now I am in the middle of a career crisis), I'll pick it up and complete it asap.

**Comment by deerskindoll** [comment]
> @deerskindoll . Thanks for the hints. I can understand. Will fix those issues. Give me a day or two (as right now I am in the middle of a career crisis), I'll pick it up and complete it asap.

Take you time, there's no pressure here.

**Comment by tolusha** [comment]
Hello @maheshrajrp 
How are you?

**Comment by maheshrajrp** [comment]
Hi @tolusha ,
I've made some changes to the PR. Can we trigger build now to validate ?

**Comment by tolusha** [comment]
@deerskindoll 
Could you have a look please?

**Comment by deerskindoll** [comment]
> @deerskindoll Could you have a look please?

The pipeline ran successfully and PR preview is available now. I noticed that in the navigation, the articles are organized like this:

image

That's not correct, is it? Virtual kubernetes installation, cloud installation and local installation are 3 separate processes on the same level, aren't they?

**Comment by maheshrajrp** [comment]
> > @deerskindoll Could you have a look please?
> 
> The pipeline ran successfully and PR preview is available now. I noticed that in the navigation, the articles are organized like this:
> 
> image
> 
> That's not correct, is it? Virtual kubernetes installation, cloud installation and local installation are 3 separate processes on the same level, aren't they?

Yes, those need to separated. If you guys could give an hint about what went wrong, I'll fix it. I did some comparison and still fail to identify the root issue. It would be great if you could help on that. Thanks

File: modules/administration-guide/nav.adoc [hunk 1]
```diff
@@ -15,6 +15,7 @@
 *** xref:user-workspaces.adoc[]
 ** xref:calculating-che-resource-requirements.adoc[]
 * xref:installing-che.adoc[]
+* xref:installing-che-on-the-virtual-kubernetes-cluster.adoc[]
```
Comment (tolusha) [inline modules/administration-guide/nav.adoc]:
```suggestion
** xref:installing-che-on-the-virtual-kubernetes-cluster.adoc[]
```

**Comment by deerskindoll** [comment]
> @deerskindoll Could you have a look please?

I'm on it but I need some time to figure out how to best present the procedure. I'll try splitting it into 2-3 smaller chunks for better readability.

**Comment by tolusha** [comment]
@maheshrajrp 
Sounds good for me.

@deerskindoll

File: modules/administration-guide/pages/installing-che-on-the-virtual-kubernetes-cluster.adoc [hunk 2]
```diff
@@ -0,0 +1,683 @@
...
+:keywords: installing-{prod-short}-on-the-virtual-kubernetes-cluster
+:page-aliases:
+
+[id="installing-{prod-short}-on-the-virtual-kubernetes-cluster.adoc"]
+= Installing {prod-short} on the virtual {kubernetes} cluster
+
+
+Virtual clusters are fully working {kubernetes} clusters that run on top of other  {kubernetes} clusters. Compared to fully separate "real" clusters, virtual clusters reuse worker nodes and networking of the host cluster. They have their own control plane and schedule all workloads into a single {namespace} of the host cluster. Similar to virtual machines, virtual clusters partition a single physical cluster into multiple separate ones.
+
+`vCluster` offers a solution for deploying {prod-short} in scenarios where it normally would be impossible. This includes situations where the {kubernetes} cluster version is incompatible with {prod-short}, or when the Cloud Service Provider lacks support for external OIDC. Utilizing `vClusters` allows bypassing these constraints and enables {prod-short} to function as needed.
```
Comment (deerskindoll) [inline modules/administration-guide/pages/installing-che-on-the-virtual-kubernetes-cluster.adoc]:
```suggestion
`vCluster` offers a solution for deploying {prod-short} in scenarios where it normally would be impossible. This includes situations where the {kubernetes} cluster version is incompatible with {prod-short}, or when the cloud service provider lacks support for external OIDC. Utilizing `vClusters` allows bypassing these constraints and enables {prod-short} to function as needed.
```

**Review by deerskindoll — COMMENTED** [review]
last small change.

**Review by deerskindoll — APPROVED** [review]
the procedure is very complex and not easy to split, let's keep it as is.

**Comment by maheshrajrp** [comment]
@tolusha @deerskindoll made the requested change.

**Review by tolusha — APPROVED** [review]

**Comment by tolusha** [comment]
@maheshrajrp 
Thank you a lot for the contribution.
We really appreciate that.

**Comment by maheshrajrp** [comment]
> @maheshrajrp 
> Thank you a lot for the contribution.
> We really appreciate that.

Thanks for merge @tolusha. And sorry for the delay.. got caught in middle of  some picky situations.. 

Additionally, I'll look into contributing to a few other documentations, too.
 
Thanks for your review and support !! Was insightful..

**Pull request merged**